cargo-dist/src/config/v0.rs

```diff
@@ -784,6 +791,9 @@ impl DistMetadata {
         if mac_pkg_config.is_none() {
             mac_pkg_config.clone_from(&workspace_config.mac_pkg_config);
         }
+        if minimum_glibc_version.is_none() {
```

Comment:
i am in favor of per package, but open to arguments against.

Reply:
Given that we're building software on the same machine I'd imagine every package will usually get the same glibc - but I can imagine highly specific environments with weird builds that would result in different versions per-package? So yeah, setting it per-package seems fine.

Reply:
yeah my reasoning here comes from the fact that this is a config to be used sparingly, usually when something is already *known* to be weird or has gone terribly wrong ;)

Reply:
Yep, that's a very good call.

Reply:
This was a deliberate choice, because it seems like there may be a benefit to supporting both.

The thing prompting this is Python tools using [`manylinux` (PEP 600)](https://peps.python.org/pep-0600/) tags. E.g., `manylinux2014` would be glibc 2.17+, which is older than cargo-dist's default.

If a workspace has multiple Python package with native extensions and they all target (e.g.) `manylinux2014`, the glibc version would be the same, so it makes sense to be able to specify it at the workspace level.

However, if someone wants to migrate those same packages to `manylinux_2_31_x86_64` (glibc 2.31, cargo-dist default) supporting it in both places would allow them to upgrade each package independently, instead of having to do it all at once.